src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/AddAzureRmSqlServerTransparentDataEncryptionCertificate.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            return null;
+        }
+
+        /// <summary>
+        /// Constructs the model to send to the update API
+        /// </summary>
+        /// <param name="model">The result of the get operation</param>
+        /// <returns>The model to send to the update</returns>
+        protected override IEnumerable<Model.AzureRmSqlServerTransparentDataEncryptionCertificateModel> ApplyUserInputToModel(IEnumerable<Model.AzureRmSqlServerTransparentDataEncryptionCertificateModel> model)
+        {
+            // Construct a new entity so we only send the relevant data to the server
+            List<Model.AzureRmSqlServerTransparentDataEncryptionCertificateModel> newEntity = new List<Model.AzureRmSqlServerTransparentDataEncryptionCertificateModel>();
+            newEntity.Add(new Model.AzureRmSqlServerTransparentDataEncryptionCertificateModel()
+            {
+                ResourceGroupName = this.ResourceGroupName,
+                ServerName = this.ServerName,
+                PrivateBlob = this.PrivateBlob,
+                Password = this.Password
+            });
+            return newEntity;
+        }
+
+        /// <summary>
+        /// Entry point for the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            switch (ParameterSetName)
+            {
+                case InputObjectParameterSet:
+                    this.ResourceGroupName = SqlServer.ResourceGroupName;
+                    this.ServerName = SqlServer.ServerName;
+                    break;
+                case ResourceIdParameterSet:
+                    var resourceInfo = new ResourceIdentifier(SqlServerResourceId);
+                    this.ResourceGroupName = resourceInfo.ResourceGroupName;
+                    this.ServerName = resourceInfo.ResourceName;
+                    break;
+                default:
+                    break;
+            }
+
+            base.ExecuteCmdlet();
+        }
+
+        /// <summary>
+        /// Sends the Firewall Rule update request to the service
+        /// </summary>
+        /// <param name="entity">The update parameters</param>
+        /// <returns>The response object from the service</returns>
+        protected override IEnumerable<Model.AzureRmSqlServerTransparentDataEncryptionCertificateModel> PersistChanges(IEnumerable<Model.AzureRmSqlServerTransparentDataEncryptionCertificateModel> entity)
+        {
+            ModelAdapter.AddAzureRmSqlServerTransparentDataEncryptionCertificate(entity.First());
+            return entity;
+        }
+
+        /// <summary>
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
+        /// </summary>
+        /// <returns>True if the model object should be written out, False otherwise</returns>
+        protected override bool WriteResult() { return false; }
```

Comment:
@nivimsft same comment about what this function does

Reply:
AzureSqlCmdletBase has a method to persist changes that returns a model. Since our cmdlet does not return any value. We return false from WriteResult() since we do not want the model object written out. The return value of WriteResult determines if the object returned by persist changes. This is called in the AzureSqlCmdletBase class.

Reply:
@nivimsft so nothing is written to the output stream by this cmdlet?